### ISSUE TITLE
clarify windows config

### ISSUE THREAD

**Issue by 3052** [kugland/neocities-deploy#66]
should be:

~~~ps1
$env:appdata + '\neocities-deploy\config\config.toml'
~~~

https://github.com/kugland/neocities-deploy/blob/60f825ede87381b681701d959a21b87e7d0048b6/README.md?plain=1#L64-L66

**Comment by kugland** [comment]
@3052 Can you take a look at https://github.com/kugland/neocities-deploy/blob/1d1d1ce62b39c03d12e376a30415251c7a21db6e/README.md. Do you think this is good?

I’m not a Windows user, but looking at it now, it seems that I should be using the local config dir, not the roaming config dir. What do you think?

**Comment by 3052** [comment]
personally I think it would be fine to remove that extra config dir:

~~~
C:\Users\Alice\AppData\Roaming\neocities-deploy\config.toml.
~~~

but otherwise yeah, its correct:

https://go.dev/pkg/os?m=old#UserConfigDir

**Comment by kugland** [comment]
@3052 you’re referecing a Go package, but this program is written in Rust. According to the documentation of the package I use, the config dir includes a `\config\config.toml`. Can you confirm that in you configuration?

**Comment by kugland** [comment]
@3052, by the way, any further improvement suggestion would be welcomed, especially on the usability on Windows and macOS platforms.

**Comment by 3052** [comment]
> you’re referecing a Go package, but this program is written in Rust.

that doesnt matter, just look at the text itself:

> On Windows, it returns `%AppData%`

> According to the documentation of the package I use, the config dir includes a `\config\config.toml`. Can you confirm that in you configuration?

according to your own docs thats not the case:

https://github.com/kugland/neocities-deploy/blob/60f825ede87381b681701d959a21b87e7d0048b6/README.md?plain=1#L64-L66

> What I mean is using the `Local` directory in `AppData`, instead of the `Roaming`, for the case of users which have multiple computers sharing the same configuration (I’m not sure how that would work on Windows, but I think that’s the difference between `Local` and `Roaming`).

again, just look at the text of the link I put:

> On Windows, it returns `%AppData%`

which on my computer is:

~~~
> $env:appdata
C:\Users\spenny\AppData\Roaming
~~~

**Comment by kugland** [comment]
Yes. However, this library I use appends `\config\` to the `Roaming` directory, for the function `config_dir`. See https://github.com/dirs-dev/directories-rs/blob/main/src/win.rs#L75

![image](https://github.com/user-attachments/assets/4a6da18a-523a-4ae5-bfe0-8237559c4c56)

**Comment by 3052** [comment]
thats absolutely wrong.

on windows cache/data dir is:

 %LocalAppData%

config dir is:

%AppData%

even for linux its wrong. it would be:

$HOME/.cache

and

$HOME/.config

https://specifications.freedesktop.org/basedir-spec/latest/

**Comment by kugland** [comment]
For Linux, it’s not wrong. See https://github.com/dirs-dev/directories-rs/blob/main/src/lin.rs#L13

It first tries to use `$XDG_CONFIG_HOME`, and if that is not defined, it uses `$HOME/.config`, which is exactly correct according to FreeDesktop’s specification:

> $XDG_CONFIG_HOME defines the base directory relative to which user-specific configuration files should be stored. If $XDG_CONFIG_HOME is either not set or empty, a default equal to $HOME/.config should be used.

**Comment by 3052** [comment]
`.config` is not `config`

**Comment by kugland** [comment]
Please be more clear. Anyway, if there was a bug here, it would be in the `directories-rs` library, not in the code of this repo.

**Comment by kugland** [comment]
> personally I think it would be fine to remove that extra config dir:
> 
> ```
> C:\Users\Alice\AppData\Roaming\neocities-deploy\config.toml
> ```
> 
> but otherwise yeah, its correct:
> 
> https://go.dev/pkg/os?m=old#UserConfigDir

Take a look at your `AppData\Roaming` directory. Even if I don’t use Windows for years now, I remember that the standard practice is for programs to create subdirectories inside it. The Go library you reference gives the base directory for configs, not the actual full path that the program should use. You can see that even in the example they give:

![image](https://github.com/user-attachments/assets/e3e7e570-a1d3-4e86-873a-ac7c2791c821)

`configPath = filepath.Join(dir, "ExampleUserConfigDir", "example.conf")` will give you something like `C:\Users\username\AppData\Roaming\ExampleUserConfigDir\example.conf`.

**Comment by 3052** [comment]
currently the configuration file is here on Windows:

~~~
$env:appdata + '\neocities-deploy\config\config.toml'
~~~

this is not "wrong", buts its not ideal. ideal would be:

~~~
$env:appdata + '\neocities-deploy\config.toml'
~~~

this is my current items:

~~~
PS C:\Users\Steven\AppData\Roaming\neocities-deploy> fd
config
config\config.toml
~~~

why add an extra folder for literally one item?

**Comment by kugland** [comment]
While I agree that there’s no point adding the extra `\config\`, this behaviour comes from an external crate. I won’t change it for two reasons: 1) it would add a conditional code path for Windows, which I can’t test locally, and, more importantly, 2) it would be a breaking change for someone already using it on Windows, the program would stop working for them.

**Comment by 3052** [comment]
OK fine, but this is still not correct:

https://github.com/kugland/neocities-deploy/blob/60f825ede87381b681701d959a21b87e7d0048b6/README.md?plain=1#L64-L66

**Comment by kugland** [comment]
I just merge this on devel, I’ll be releasing a new version to master in a few minutes.

**Comment by kugland** [comment]
Closed with fccdfbd6904fbb952f32ffcec721ae32ceecccff.
